test/integration/test_utils.mjs

```diff
@@ -100,23 +112,29 @@ function getSelectedEditors(page) {
 }
 
 async function waitForEvent(page, eventName, timeout = 5000) {
-  const hasTimedout = await Promise.race([
-    // add event listener and wait for event to fire before returning
-    page.evaluate(
-      name =>
-        new Promise(resolve => {
-          document.addEventListener(name, () => resolve(false), { once: true });
-        }),
-      eventName
-    ),
-    page.evaluate(
-      timeOut =>
-        new Promise(resolve => {
-          setTimeout(() => resolve(true), timeOut);
-        }),
-      timeout
-    ),
-  ]);
+  const handle = await page.evaluateHandle(
+    (name, timeOut) => {
+      let callback = null;
+      return [
+        Promise.race([
+          new Promise(resolve => {
+            // add event listener and wait for event to fire before returning
+            callback = () => resolve(false);
+            document.addEventListener(name, callback, { once: true });
+          }),
+          new Promise(resolve => {
+            setTimeout(() => {
+              document.removeEventListener(name, callback);
+              resolve(true);
+            }, timeOut);
+          }),
+        ]),
+      ];
+    },
+    eventName,
+    timeout
+  );
+  const hasTimedout = await handle.evaluate(([p]) => p);
```

Comment:
Nit: For overall consistency, shouldn't this use the new helper function instead?
```suggestion
  const hasTimedout = await awaitPromise(handle);
```